test/fiuba/algo3/tp2/juegoTest/MapaTest.java

```diff
@@ -38,4 +40,39 @@ public void alAlmacenarElementoEnCasilleroVacioExisteElemento() {
         Assert.assertEquals(true, mapa.existeElementoEnPosicion(unaPosicion));
     }
 
+    @Test
+    public void sePuedeOcuparUnCasilleroVacioDelTerreno() {
+        Mapa mapa = new Mapa(20,20);
+        Navegador navegador = new Navegador(5,5, mapa);
+        Jugador jugador = new Jugador(navegador);
+        Posicion unaPosicionAlNorte = new Posicion(5,6);
+        mapa.vaciarContenidoCasillero(unaPosicionAlNorte);
+        jugador.moverNorte();
+
+        Assert.assertEquals(unaPosicionAlNorte.getX(), navegador.getPosicionActual().getX());
+        Assert.assertEquals(unaPosicionAlNorte.getY(), navegador.getPosicionActual().getY());
+    }
+
+    @Test
+    public void noSePuedeOcuparUnCasilleroOcupadoDelTerreno() {
+        Mapa mapa = new Mapa(20,20);
+        Navegador navegador = new Navegador(5,5, mapa);
+        Jugador jugador = new Jugador(navegador);
+        Posicion unaPosicionAlNorte = new Posicion(5,6);
+        MaderaMaterial unaMadera = new MaderaMaterial();
+        mapa.setContenidoCasillero(unaMadera, unaPosicionAlNorte);
+        jugador.moverNorte();
```

Comment:
esto seria un momento ideal para poder introducir excepciones

Reply:
Correcto, habra que ver como se hacen en jaja

Reply:
en java of course